cpp/proto/xjson_serializer.cc

```diff
@@ -41,7 +41,12 @@ SerializeResult CheckJsonFormat(const string& json) {
 
 string V1LeafData(const LogEntry& entry) {
   CHECK(entry.has_x_json_entry());
-  return entry.x_json_entry().json();
+  hash h;
+  if (!common_json_hash(entry.x_json_entry().json().c_str(), h)) {
+	LOG(ERROR) << "Faild to compute object hash: "
+	           << entry.x_json_entry().json() << ".";
+  }
+  return string(reinterpret_cast<const char*>(&h[0]), sizeof(h));
```

Comment:
Indeed it is quite different than the others.  I couldn't find a better place in the libraries to use object hash.  The typical pattern is Sha256(LeafData(entry)).  Where LeafData is the function you can swap out.   There's no current support for defining some function LeafHash that does this differently.